src/settings/options.rs

```diff
@@ -251,6 +257,7 @@ pub struct Options {
             select = ["E", "F", "B", "Q"]
         "#
     )]
+    /// A list of check code prefixes to enable.
```

Comment:
I think it's unfortunate if we have to repeat ourselves here. I would look into whether we can adjust the `option` macro to attach the `doc` attribute to the field (would probably be nice during development as well), or whether we can make use of the generated `ConfigurationOptions` struct that exposes all that information.

Reply:
I agree. I'll look into it

Reply:
We can also explore turning this around: see how far we can get with JSON schema & schemars, and perhaps we can build on that as a base.

Reply:
Repeating ourselves here isn't the end of the world (it's nice to have these as rust docs too, I guess?) though agree it'd be nice to reuse if there's a straightforward path.

Reply:
Yeah, there's a bit of overlap between the two traits so at some point (not necessarily in this PR) a single source of truth should be in place.

Reply:
Okay, I think I have this working in the other direction?

Like we can do this:

```rust
#[option(
    doc = r#""#,
    default = r#"[]"#,
    value_type = "Vec<char>",
    example = r#"
        # Allow minus-sign (U+2212), greek-small-letter-rho (U+03C1), and the asterisk-operator (U+2217),
        # which could be confused for "-", "p", and "*", respectively.
        allowed-confusables = ["−", "ρ", "∗"]
    "#
)]
/// A list of allowed "confusable" Unicode characters to ignore when enforcing `RUF001`,
/// `RUF002`, and `RUF003`.
pub allowed_confusables: Option<Vec<char>>,
...
```

...and have it generate the schema, and the documentation, and it works in the IDE.

What do you think @squiddy?

Reply:
(It's probably possible to generate the rustdoc from the proc macro, but I'm not that strong on macros (and I don't know how that would ever work with the IDE.))

Reply:
Yes, looks good to me! As you said, that fit's more into standard rust and how IDEs work. I found some hints on how to generate rustdoc from the macro, but than I ran into macro errors that I couldn't resolve.. probably for the best.

(On that topic, I wonder if we can just move everything into rustdoc and have the macro parse that text up into `default` / `value_type` and `example`.)